lib/presentation/widget/upload_request_inside/upload_request_inside_viewmodel.dart

```diff
@@ -132,4 +155,178 @@ class UploadRequestInsideViewModel extends BaseViewModel {
+    });
+  }
+
+  void _dispatchHandleDownloadAction(List<UploadRequestEntry> entries) {
+    store.dispatch(_handleDownloadEntries(entries));
+  }
+
+  OnlineThunkAction _handleDownloadEntries(List<UploadRequestEntry> entries) {
+    return OnlineThunkAction((Store<AppState> store) async {
+      await _downloadEntriesInteractor.execute(entries).then((result) =>
+        result.fold(
+          (failure) => store.dispatch(UploadRequestInsideAction(Left(failure))),
+          (success) => store.dispatch(UploadRequestInsideAction(Right(success)))));
+    });
+  }
+
+  void _showDownloadingFileDialog(BuildContext context, List<UploadRequestEntry> entries, CancelToken cancelToken) {
+    final downloadMessage = entries.length <= 1
+      ? AppLocalizations.of(context).downloading_file(entries.first.name)
+      : AppLocalizations.of(context).downloading_files(entries.length);
+    showCupertinoDialog(
+      context: context,
+      builder: (_) => DownloadingFileBuilder(cancelToken, _appNavigation)
+        .key(Key('downloading_file_dialog'))
+        .title(AppLocalizations.of(context).preparing_to_export)
+        .content(downloadMessage)
+        .actionText(AppLocalizations.of(context).cancel)
+        .build());
+  }
+
+  OnlineThunkAction _exportFileAction(List<UploadRequestEntry> entries, CancelToken cancelToken) {
+    return OnlineThunkAction((Store<AppState> store) async {
+      await _downloadMultipleEntryIOSInteractor
+        .execute(entries, cancelToken)
+        .then((result) => result.fold(
+          (failure) => store.dispatch(_exportFileFailureAction(failure)),
+          (success) => store.dispatch(_exportFileSuccessAction(success))));
+    });
+  }
+
+  ThunkAction<AppState> _exportFileSuccessAction(Success success) {
+    return (Store<AppState> store) async {
+      _appNavigation.popBack();
+      if (success is DownloadEntryIOSViewState) {
+        await share_library.Share.shareFiles(
+            [success.filePath]);
+      } else if (success is DownloadEntryIOSAllSuccessViewState) {
+        await share_library.Share.shareFiles(success.resultList
+            .map((result) => ((result.getOrElse(() => IdleState()) as DownloadEntryIOSViewState).filePath))
+            .toList());
+      } else if (success is DownloadEntryIOSHasSomeFilesFailureViewState) {
+        await share_library.Share.shareFiles(success.resultList
+            .map((result) => result.fold(
+                (failure) => '',
+                (success) => ((success as DownloadEntryIOSViewState).filePath)))
+            .toList());
+      }
+    };
+  }
+
+  ThunkAction<AppState> _exportFileFailureAction(Failure failure) {
+    return (Store<AppState> store) async {
+      if (failure is DownloadEntryIOSFailure
+          && !(failure.downloadFileException is CancelDownloadFileException)) {
+        _appNavigation.popBack();
+      }
+      store.dispatch(UploadRequestInsideAction(Left(failure)));
+    };
+  }
+
+  @override
+  void onDisposed() {
```

Comment:
don't you need to clear upload request inside state ? idk if you need it outside this view

Reply:
right, missing it